api/src/Security/JWT/ValidatingSplitCookieExtractor.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+namespace App\Security\JWT;
+
+use Lexik\Bundle\JWTAuthenticationBundle\TokenExtractor\TokenExtractorInterface;
+use Symfony\Component\HttpFoundation\Request;
+
+class ValidatingSplitCookieExtractor implements TokenExtractorInterface {
+    public function __construct(private TokenExtractorInterface $decorated) {
+    }
+
+    public function extract(Request $request) {
+        $extractedTokens = $this->decorated->extract($request);
+        if (!$extractedTokens) {
+            return false;
+        }
+        $parts = explode('.', $extractedTokens);
+        if (sizeof($parts) < 3) {
+            return false;
+        }
```

Comment:
This fixes our specific problem with the logout for now, but not the general case of some cookies missing. When e.g. the jwt_s cookie is missing for some reason, there will still be two periods in the reconstructed token: `eyHeader.eyPayload.`

In the current state, this class is only a temporary fix until we get a more complete fix into the Lexik JWT bundle. If we plan on keeping this class long term (or if we assume that a fix won't be accepted upstream), I'd generalize this implementation by asserting e.g. that none of the parts may be empty, or by directly validating that both cookies are present on the request.

What do you think?

Reply:
no, because the implementation of walks through empty elements and removes them, if the cookie exists, but was empty or does not exist. So no empty Strings are imploded with dots.
The case with a period at the end currently does not happen.

I agree that it makes no sense that we verify the concatenated cookie, when the information about the single token parts where available before.
If we want to validate more strictily (test if the specific cookies are there), then we should replace the class with our own implementation.

Reply:
done